### PR TITLE
Make PageField only list pages in the same language as the edited page

### PR DESCRIPTION
**Pull request by sephii**
This is a workaround for #3981. It introduces a new `CMSPluginBase` derived class named `CMSPagePluginBase`, that, if inherited, will make `PageField` fields only list pages in the same language as the edited page. I think it would be better if it could be included in `CMSPluginBase` since `PageField` is still a core feature.

If we leave it as `CMSPagePluginBase`, then this should be documented somewhere.

### CODE REVIEW

**Comment by sephii** [comment]
Please note that the tests are currently failing, mainly because of the check on the number or executed SQL queries. I'm not sure how to fix this.

**Comment by yakky** [comment]
@sephii for number of queries just update the allowed range. It's normal that query number is larger whenver you add plugins to the test suite (like in this case) or if the plugins does more queries

**Comment by yakky** [comment]
I'll review and check whether the new mixin class is really needed.

File: cms/plugin_base.py [hunk 1]
```diff
@@ -361,6 +361,15 @@ def get_plugin_media(self, request, context, plugin):
         )
 
 
+class CMSPagePluginBase(CMSPluginBase):
+    def formfield_for_dbfield(self, db_field, **kwargs):
+        from cms.models import fields
+        if isinstance(db_field, fields.PageField):
+            kwargs['language'] = self.cms_plugin_instance.language
```
Comment (czpython) [inline cms/plugin_base.py]: cms_plugin_instance is no longer used when adding a plugin on 3.3

**Comment by czpython** [comment]
@sephii 
Hello,
Thanks for the contribution.
I've reviewed the PR and seems to me that adding the logic to the plugin class couples plugins with the page field.

While agree that usually it's desired to see the pages available for the plugin the field is used on, I think it's the responsibility of the plugin author to set this up.
We can improve the plugin and page field internals to make it easier for a plugin author to filter pages based on the plugin's language. /cc @divio/django-cms-core

**Pull request closed without merging**